modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -257,17 +266,54 @@ export default class Ecdsa {
     return derivedPk + derivedChaincode;
   }
 
+  /**
+   * Generate a range proof challenge and add it to shares previously created
+   * by #keyCombine.
+   * @param {XShare} xShare Private xShare of signer
+   * @param {YShare} yShare YShare of the other participant involved in
+   * this signing operation
+   * @returns {KeyCombined} The new XShare and YShares with the amended
+   * challenge values
+   */
+  signChallenge(xShare: XShare, yShare: YShare): KeyCombinedWithNTilde {
+    const challenge = rangeProof.generateNTilde(3072);
+    const ntilde = bigIntToBufferBE(challenge.ntilde, 384).toString('hex');
+    const h1 = bigIntToBufferBE(challenge.h1, 384).toString('hex');
+    const h2 = bigIntToBufferBE(challenge.h2, 384).toString('hex');
+    const shares = {
+      xShare: { ...xShare, ntilde, h1, h2 },
+      yShares: {
+        [yShare.j]: {
+          i: yShare.j,
+          j: yShare.i,
+          n: xShare.n,
+          ntilde,
+          h1,
+          h2,
+        },
+      },
+    };
+    return shares;
+  }
+
   /**
    * Create signing shares.
    * @param {xShare} xShare Private xShare of current participant signer
    * @param {YShare} yShare yShare corresponding to the other participant signer
    * @returns {SignShareRT} Returns the participant private w-share
    * and k-share to be distributed to other participant signer
    */
-  signShare(xShare: XShare, yShare: YShare): SignShareRT {
+  signShare(xShare: XShare | XShareWithNTilde, yShare: YShare | YShareWithNTilde): SignShareRT {
     const pk = getPaillierPublicKey(hexToBigInt(xShare.n));
 
+    // Generate a challenge if ntilde is not present in the xShare.
+    if (!hasNTilde(xShare)) {
```

Comment:
Can you help me understand under what case would this happen?

Reply:
If A and B are signing, A can call signChallenge and send her challenge y share to B. This is a shortcut for B to call signShare directly using his challengeless x-share without calling signChallenge beforehand.